bookstore/tests/test_clone.py

```diff
@@ -301,3 +302,89 @@ def test_validate_relpath_escape_basedir(caplog):
         with caplog.at_level(logging.INFO):
             fs_clonepath = validate_relpath(relpath, settings, log)
     assert f"Request to clone from a path outside of base directory" in caplog.text
+
+
+class TestFSCloneHandler(AsyncTestCase):
+    def setUp(self):
+        super().setUp()
+        mock_settings = {"BookstoreSettings": {"fs_cloning_basedir": "/Users/jupyter"}}
+        self.config = Config(mock_settings)
+        self.mock_application = Mock(
+            spec=Application,
+            ui_methods={},
+            ui_modules={},
+            settings={'jinja2_env': Environment(), 'config': self.config},
+        )
+
+    def get_handler(self, uri, app=None):
+        if app is None:
+            app = self.mock_application
+        connection = Mock(context=Mock(protocol="https"))
+        payload_request = HTTPRequest(
+            method='GET',
+            uri=uri,
+            headers={"Host": "localhost:8888"},
+            body=None,
+            connection=connection,
+        )
+        return BookstoreFSCloneHandler(app, payload_request)
+
+    @gen_test
+    async def test_get_no_param(self):
+        empty_handler = self.get_handler('/bookstore/fs-clone')
+        with pytest.raises(HTTPError):
+            await empty_handler.get()
+
+    @gen_test
+    async def test_get_empty_relpath(self):
+        no_bucket_handler = self.get_handler('/bookstore/fs-clone?relpath=')
+        with pytest.raises(HTTPError):
+            await no_bucket_handler.get()
+
+    @gen_test
+    async def test_get_escape_basedir(self):
+        no_bucket_handler = self.get_handler('/bookstore/fs-clone?relpath=../hi')
```

Comment:
name seems misleading

Reply:
The perils of copying other code as a template 😛